### ISSUE TITLE
What to do with the current icons in the icons package

### ISSUE THREAD

**Issue by HendrikThePendric** [dhis2/ui#305]
Whilst writing a comment in the icon component API discussion (issue #285) I started thinking about what to do with the icons already present. My thoughts can be summarised like so:
- I thought that the icon library was an internal thing and only used by components in `core` and `widgets`, but unfortunately it's not. We can find it [on NPM](https://www.npmjs.com/package/@dhis2/ui-icons).
- This makes removing the current icons a breaking change
- I think we need to press ahead and remove them anyway, because it just doesn't seem like these icons really belong in an icon library: they are closely tied to the component and or are too opinionated.
- I think we should remove these icons and then in `core` and `widgets` go through all the instances where we import an icon and choose one of these options:
    1. If the icon doesn't have any additional behaviour and is included/has an equivalent in the svg-icon-set, we should use that instead.
    2. If the icon has additional behaviour it should probably be colocated with the component that uses it, or be implemented as a standalone component if multiple components rely on it.
    3. If the icon doesn't have any additional behaviour and doesn't have an equivalent in the svg-icon-set, then we need to discuss with @cooper-joe what to do. Most likely course of action would be that the icon gets added to the svg-icon-set.

Does this make sense, or should we do things differently?

**Comment by ismay** [comment]
I think we discussed that a while back, but I'm not sure what the outcome was. I think we discussed it when moving to the monorepo setup.

**Comment by cooper-joe** [comment]
I agree with the three options you outline @HendrikThePendric. Looking through the list of current icons, I think the vast majority are covered by there. There will be a few that will require discussion about whether it should live in `icons` or with the component, but I don't foresee any big issues.
